### PR TITLE
Add artifact removal modes (linear & cubic interpolation)

### PR DESCRIPTION
**Pull request by joramvanrheede**
Add 'mode' input argument to remove_artifacts(). Default is 'zeros' (old behaviour). If 'linear', implements linear interpolation between pre and post artifact points. If 'cubic' - implements cubit spline interpolation.
Additional input 'fit_sample_spacing' determines how far apart sample points are taken for creating the spline fit that is used to fill in the gap (default is 1 ms). Shorter sample intervals lead to better continuation locally, but may lead to steeper fit slopes with more high frequency content.

Related to issue #451

### CODE REVIEW

File: spiketoolkit/preprocessing/remove_artifacts.py [hunk 1]
```diff
@@ -22,21 +25,113 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
...
+                else:
+                    # No data to interpolate from on either side of gap;
+                    # Fill with zeros
+                    traces[:,gap_idx] = 0
+                    
         return traces
 
 
-def remove_artifacts(recording, triggers, ms_before=0.5, ms_after=3):
+def remove_artifacts(recording, triggers, ms_before=0.5, ms_after=3, mode = 'zeros', fit_sample_spacing = 1):
```
Comment (alejoe91) [inline spiketoolkit/preprocessing/remove_artifacts.py:134]: @joramvanrheede Can you add some explanation of the different modes in the docstrings? also `fit_sample_spacing` should be added there :)

Comment (joramvanrheede) [reply spiketoolkit/preprocessing/remove_artifacts.py:134]: Yes, will do that now. I also see a 'Travis CI build failed' notification below but I'm unfamiliar with debugging this sort of issue...

Comment (alejoe91) [reply spiketoolkit/preprocessing/remove_artifacts.py:134]: I updated some stuff on the extractors side and this makes the test failing. I'll fix that. Thanks for the additions!

Give me some other few days to properly test this. In the meanwhile, could you add some mock tests here: https://github.com/SpikeInterface/spiketoolkit/blob/master/spiketoolkit/tests/test_preprocessing.py#L197-L215 ?
Basically similar to what's there already but adding the new option

**Comment by joramvanrheede** [comment]
Hi Alessio, I see you've already sorted the tests now - Thank you for that! I had a look last week but wasn't immediately sure what tests would be appropriate (for the zeroed out artifacts it made sense to check everything was zero but i wasn't sure what would be a good equivalent for the cubic and linear fits, without re-creating the entire fit on a local bit of trace in the test).
I also saw you were more efficient with the scipy import and fixed some code style issues - thanks! This is my first time contributing to a public repository that is not just my own so I'm still learning about conventions...

**Comment by alejoe91** [comment]
No worries Joram! Just made a few changes :) thanks for the contribution! Looking forward to more!

**Pull request merged**